src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.NoCom.cs

```diff
@@ -35,9 +36,8 @@ public static void CleanupUnusedObjectsInCurrentContext()
         {
         }
 
-        public static IntPtr CreateAggregatedObject<T>(IntPtr pOuter, T o)
+        public static IntPtr CreateAggregatedObject<T>(IntPtr pOuter, T o) where T : object
```

Comment:
Why using a constraint here instead of DisallowNullAttribute? Just curious.

Reply:
Same as earlier: my preference is to use the annotations / constraints wherever possible, and then only fall back to the attributes when things can't be expressed fully with the annotations alone.